lib/converters/utils.test.js

```diff
@@ -17,6 +17,517 @@ function expectEntity(entity, entity_id, id, name, type) {
+                const entityType = "does-not-matter";
+
+                const obj = {
+                    _id: "ioBroker object ID",
+                };
+
+                const entity = utils.processCommon(null, null, null, obj, entityType);
+
+                expect(entity.entity_id).to.equal(
+                    `${entityType}.${obj._id.replace(/\s/g, "_")}`
+                );
+            });
+
+            it("from common.name", function () {
+                const entityType = "does-not-matter";
+
+                const obj = {
+                    common: {
+                        name: "a common name",
+                    },
+                    _id: "ignored",
+                };
+
+                const entity = utils.processCommon(null, null, null, obj, entityType);
+
+                expect(entity.entity_id).to.equal(
+                    `${entityType}.${obj.common.name.replace(/\s/g, "_")}`
+                );
+            });
+
+            it("from English common.name", function () {
+                const entityType = "does-not-matter";
+
+                const obj = {
+                    common: {
+                        name: {
+                            en: "a common name",
+                            de: "ignored",
+                        },
+                    },
+                    _id: "ignored",
+                };
+
+                entityData.lang = "de";
+
+                const entity = utils.processCommon(null, null, null, obj, entityType);
+
+                expect(entity.entity_id).to.equal(
+                    `${entityType}.${obj.common.name.en.replace(/\s/g, "_")}`
+                );
+            });
+
+            it("from instance-specific name", function () {
+                const entityType = "does-not-matter";
+
+                const obj = {
+                    common: {
+                        name: "ignored",
+                        custom: {
+                            [entityData.adapter.namespace]: {
+                                name: "instance-specific name",
+                            },
+                        },
+                    },
+                    _id: "ignored",
+                };
+
+                const entity = utils.processCommon(
+                    null,
+                    "ignored",
+                    "ignored",
+                    obj,
+                    entityType
+                );
+
+                expect(entity.entity_id).to.equal(
+                    `${entityType}.${obj.common.custom[
+                        entityData.adapter.namespace
+                    ].name.replace(/\W/g, "_")}`
+                );
+            });
+
+            xit("from predefined entity ID argument", function () {
+                const entityType = "does-not-matter";
+                const predefinedEntityId = "predefined id";
+
+                const obj = {
+                    common: {
+                        // TODO
```

Comment:
@Garfonso This is the third TODO for your review. The diff is collapsed by default, which is why you probably did not find it.